packages/crw-client/src/client.rs

```diff
@@ -26,24 +24,37 @@ impl CosmosClient {
     /// The client will communicate using `lcd_address` for the legacy LCD requests
     /// and `grpc_addr` for the new gRPC request.
     ///
-    /// # Errors
-    /// Returns an [`Err`] if `grpc_addr` is an invalid URI.
-    ///
     ///# Examples
     ///
     /// ```
     ///  use crw_client::client::CosmosClient;
     ///
-    ///  let client = CosmosClient::new("http://localhost:1317", "http://localhost:9090").unwrap();
+    ///  let client = CosmosClient::new("http://localhost:1317", "http://localhost:9090");
     /// ```
-    pub fn new(lcd_addr: &str, grpc_addr: &str) -> Result<CosmosClient, InvalidUri> {
-        let grpc_uri = grpc_addr.parse::<Uri>()?;
-        let grpc_channel = Channel::builder(grpc_uri);
-
-        Ok(CosmosClient {
-            grpc_channel,
+    pub fn new(lcd_addr: &str, grpc_addr: &str) -> CosmosClient {
+        CosmosClient {
+            grpc_addr: grpc_addr.to_string(),
             lcd_addr: lcd_addr.to_string(),
-        })
+        }
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    pub async fn grpc_client(&self) -> Result<tonic::transport::Channel, CosmosError> {
+        let grpc_uri = self
+            .grpc_addr
+            .parse::<tonic::codegen::http::Uri>()
+            .map_err(|err| CosmosError::Grpc(err.to_string()))?;
+        let grpc_channel = tonic::transport::Channel::builder(grpc_uri);
+
+        Ok(grpc_channel
+            .connect()
+            .await
+            .map_err(|err| CosmosError::Grpc(err.to_string()))?)
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    pub async fn grpc_client(&self) -> Result<grpc_web_client::Client, CosmosError> {
```

Comment:
This can be private since we don't need to expose this outside
```suggestion
    async fn grpc_client(&self) -> Result<grpc_web_client::Client, CosmosError> {
```